src/CSharp.Workspaces/CSharp/SyntaxLogicalInverter.cs

```diff
@@ -128,13 +128,16 @@ internal static SyntaxLogicalInverter GetInstance(Document document)
             case SyntaxKind.IsExpression:
                 {
                     var isExpression = (BinaryExpressionSyntax)expression;
-
-                    return IsPatternExpression(
-                        isExpression.Left,
-                        isExpression.OperatorToken.WithTrailingTrivia(Space),
-                        UnaryPattern(
-                            Token(SyntaxKind.NotKeyword).WithTrailingTrivia(isExpression.OperatorToken.TrailingTrivia),
-                            TypePattern((TypeSyntax)isExpression.Right)));
+                    var rightTypeSymbol = (ITypeSymbol)semanticModel.GetSymbolInfo(isExpression.Right, cancellationToken).Symbol!;
+                    return (Options.UseNotPattern)
+                        ? IsPatternExpression(
+                            isExpression.Left,
+                            isExpression.OperatorToken.WithTrailingTrivia(Space),
+                            UnaryPattern(
+                                Token(SyntaxKind.NotKeyword)
+                                    .WithTrailingTrivia(isExpression.OperatorToken.TrailingTrivia),
+                                TypePattern(rightTypeSymbol.ToTypeSyntax(SymbolDisplayFormat.FullyQualifiedFormat))))
```

Comment:
I don't see Simplifier annotation being added here.

Reply:
That's because it cannot be added. https://github.com/dotnet/roslyn/issues/69426

Reply:
Please see my newer comment regarding the entire section.